simplq/src/main/java/me/simplq/service/QueueService.java

```diff
@@ -123,6 +123,10 @@ public QueueDetailsResponse getQueueDetails(String queueId) {
                   .filter(token -> token.getStatus() != TokenStatus.REMOVED)
                   .sorted(Comparator.comparing(Token::getTokenCreationTimestamp))
                   .forEach(resp::addToken);
+              queue.getTokens().stream()
+                  .filter(token -> token.getStatus() == TokenStatus.REMOVED)
+                  .sorted(Comparator.comparing(Token::getTokenCreationTimestamp))
+                  .forEach(resp::addDeletedToken);
```

Comment:
Hi. A small optimisation. Let's check in the QueueDetailsResponse::addToken for the token status, and add to the deleted or active tokens list from in the class, and then we can remove this filter here instead.